### PR TITLE
docs: add correct array signatures for Modules

### PR DESCRIPTION
**Pull request by paulbalandan**
**Description**
See #7731

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: app/Config/Modules.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ class Modules extends BaseModules
      *
      * If it is not listed, only the base application elements will be used.
      *
-     * @var string[]
+     * @var list<string>
```
Comment (kenjis) [inline app/Config/Modules.php:75]: If `list` is okay with `@var`, we can remove `@phpstan-var` tags like:
```
     * @var array<int, string>
     * @phpstan-var list<string>
```

**Comment by kenjis** [comment]
I changed `fix:` in the subject to `docs:` because when checking the changelog generated at release time, we need to think a bit whether the item is a bug fix or an item that should be removed from the changelog.
I would like to limit `fix:` to product code fixes only.

File: app/Config/Modules.php (hunk 1 above)
Comment (paulbalandan) [reply app/Config/Modules.php:75]: Interesting. I think vscode intelephense has now support for this type, as well as Closure signatures.

Comment (kenjis) [reply app/Config/Modules.php:75]: Okay, then let's remove `@phpstan-var` from now on.

**Review by kenjis — APPROVED** [review]

File: app/Config/Modules.php (hunk 1 above)
Comment (MGatner) [reply app/Config/Modules.php:75]: I'm all for this; I can confirm that Intelephense (in BBEdit) supports `list` as well.

Comment (MGatner) [reply app/Config/Modules.php:75]: I know I always ask this... these are different, right?
```
list<string>
string[]
```

I think the latter just means "values are all strings" without any indication of keys. But I can't find the answer explicitly. 

* https://phpstan.org/writing-php-code/phpdoc-types#general-arrays
* https://docs.phpdoc.org/2.9/guides/types.html#arrays

**Review by MGatner — APPROVED** [review]

**Comment by MGatner** [comment]
@paulbalandan Since it's vaguely related could you roll in the CS Fixer change to this PR?

+++ /home/runner/work/CodeIgniter4/CodeIgniter4/system/Exceptions/FrameworkException.php

File: app/Config/Modules.php (hunk 1 above)
Comment (kenjis) [reply app/Config/Modules.php:75]: Yes.
> `/** @return ValueType[] */`
> In Psalm this annotation is equivalent to @psalm-return array<array-key, ValueType>.
https://psalm.dev/docs/annotating_code/type_syntax/array_types/#phpdoc-syntax

Comment (paulbalandan) [reply app/Config/Modules.php:75]: same for phpstan

Comment (paulbalandan) [reply app/Config/Modules.php:75]: Sent PR: #8114

**Pull request merged**